Assets/Scripts/UIManager.cs

```diff
@@ -0,0 +1,139 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using UnityEngine.UI;
+
+
+public class UIManager : MonoBehaviour
+{
+    // Start is called before the first frame update
+
+
+ 
+    public TextMeshProUGUI playerOneScoreLabel;
+    public TextMeshProUGUI playerOneLivesLeftLabel;
+    public GameObject playerOneDurability;
+    public GameObject playerOneHealth;
+    public GameObject playerOneInventory;
+
+
+    public int playerOneScoreNum;
+    public int playerOneLivesLeftNum;
+    public float playerOneDurabilityScore = 0.2f;
+    public float playerOneHealthScore;
+    public string playerOneInventoryType;
+
+    //public GameObject playerTwoScore;
+    //public GameObject playerTwoDurability;
+    //public GameObject playerTwoHealth;
+    //public GameObject playerTwoInventory;
+    //public int playerTwoScoreNum;
+    //public int playerTwoDurabilityScore;
+    //public int playerTwoHealthScore;
+    //public string playerTwoInventoryType;
+
+
+
+    private void Update()
+    {
+        if (Input.GetKeyDown("z"))
```

Comment:
1. We are not using the old input system already, so please do not use these codes here.
2. You should hook these functions of updating UI to some events about players getting a score, changing health, etc.

The thing you wrote here is doing nothing in terms of gameplay.